src/client/wrappers/loadAndErrorWrapper/index.js

```diff
@@ -1,78 +1,37 @@
-import React, { Fragment, useContext } from 'react'
+import React, { Fragment } from 'react'
 import PropTypes from 'prop-types'
 
-import { BoxContext } from '../box'
 import Initializing from './Initializing'
 import LoadingPublicProfile from './LoadingPublicProfile'
 import ErrorState from './Error'
 import UnlockingBox from './UnlockingBox'
+import NoProfile from './NoProfile'
+import { useProfileStates } from '../../hooks'
 
-const LoadAndErrorWrapper = props => {
-  const syncing = false
+const LoadAndErrorWrapper = ({ children, ethereumAddress }) => {
+  const {
+    loadingPublicProf,
+    unlockingProf,
+    noPublicProfileFound,
+    error,
+  } = useProfileStates()
 
-  const { boxes } = useContext(BoxContext)
+  // it was hardcoded as syncing=false before
+  const isInitializing = false
 
-  const usersBox = boxes[props.ethereumAddress]
-
-  const loadingPublicProf = usersBox && usersBox.loadingPublicProf
-  const unlockingProf = usersBox && usersBox.unlockingProf
-
-  return (
-    <LoadAndErrorView
-      {...props}
-      isInitializing={syncing}
-      isLoadingPublicProfile={loadingPublicProf}
-      isUnlockingProfile={unlockingProf}
-    />
-  )
-}
-
-LoadAndErrorWrapper.propTypes = {
-  ethereumAddress: PropTypes.string.isRequired,
-}
-
-const LoadAndErrorView = ({
-  children,
-  isInitializing,
-  isLoadingPublicProfile,
-  isUnlockingProfile,
-  error,
-}) => {
   if (Object.keys(error).length > 0) return <ErrorState />
   if (isInitializing) return <Initializing />
-  if (isLoadingPublicProfile) return <LoadingPublicProfile />
-  if (isUnlockingProfile) return <UnlockingBox />
+  if (loadingPublicProf) return <LoadingPublicProfile />
+  if (unlockingProf) return <UnlockingBox />
+  if (noPublicProfileFound) {
```

Comment:
please remove brackets here for consistency :)

Reply:
Sure!